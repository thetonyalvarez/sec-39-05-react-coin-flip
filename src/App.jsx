import { useState } from 'react'
import logo from './logo.svg'
import Coin from './Coin'
import Counter from './Counter'
import Button from '@mui/material/Button'
import Container from '@mui/material/Container'
import './App.css'

let coin;


function App() {
  
  const flipCoin = () => {
    return Math.floor(Math.random() * 2) + 1
  }
  
  const [side, setSide] = useState(0)
  const [totalCount, setTotalCount] = useState(0)
  const [headCount, setHeadCount] = useState(0)
  const [tailCount, setTailCount] = useState(0)

  const handleFlip = () => {
    let flip = flipCoin()
    
    setTotalCount(totalCount + 1)

    if (flip == 1) {
      setHeadCount(headCount + 1)
    } else if (flip == 2) {
      setTailCount(tailCount + 1)
    }

    setSide(flip)

  }

  return (
    <div className="App">
      <header className="App-header">
        <p>Let's Flip a Coin!</p>
      </header>
      <Container component="main" sx={{py: 4}}>
        <Coin sx={{p: 4}} side={side}/>
        <Button variant="outlined" onClick={handleFlip}>FLIP ME!</Button>
        <Counter total={totalCount} heads={headCount} tails={tailCount}/>
      </Container>
    </div>
  )
}

export default App