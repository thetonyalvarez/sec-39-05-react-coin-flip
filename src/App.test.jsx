import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { expect, it, describe } from 'vitest'
import App from "./App"

it("renders without crashing", function() {
  render(<App/>)
});

it("matches snapshot", function() {
  const {asFragment} = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

it("works when you click the 'flip' button", function() {
  const { queryByTestId, getByTestId } = render(<App/>)

  const button = queryByTestId('flip-btn')
  fireEvent.click(button)
  
  const coinImg = getByTestId('coin-img')
  expect(coinImg).toBeInTheDocument();
});