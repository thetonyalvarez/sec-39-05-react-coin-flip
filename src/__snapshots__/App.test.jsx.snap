// Vitest Snapshot v1

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <p>
        Let's Flip a Coin!
      </p>
    </header>
    <main
      class="MuiContainer-root MuiContainer-maxWidthLg css-1sqq6so-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-0"
      />
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        data-testid="flip-btn"
        tabindex="0"
        type="button"
      >
        FLIP ME!
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-5z7jm9-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-3 css-1ozs4dl-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-xt7fqp-MuiPaper-root-MuiCard-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
            >
              Total Flips
            </h6>
            <h3
              class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
            >
              0
            </h3>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-6 MuiGrid-grid-lg-3 css-1eehwzh-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-xt7fqp-MuiPaper-root-MuiCard-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
            >
              Heads
            </h6>
            <h3
              class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
            >
              0
            </h3>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-6 MuiGrid-grid-lg-3 css-1eehwzh-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-xt7fqp-MuiPaper-root-MuiCard-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
            >
              Tails
            </h6>
            <h3
              class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
            >
              0
            </h3>
          </div>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
